arcgis-ios-sdk-samples/AppDelegate.swift

```diff
@@ -91,6 +95,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
         }
     }
     
+    // MARK: - Google Analytics settings
+    
+    func setGoogleAnalyticsPref() {
+        // Enable/disable GA based on settings.
+        let disableGoogleAnalytics = UserDefaults.standard.bool(forKey: "disableGoogleAnalytics")
```

Comment:
This is due to an iOS limitation, see: https://stackoverflow.com/questions/36253998

Reply:
This isn't an iOS limitation but rather a misunderstanding of the User Defaults API. It has been a while since I've used it, but if I remember quickly, default values must be registered using `UserDefaults.register(defaults:)`. Also, be aware that the setting can change while the app is active, so the setting should be observed for changes as well. In fact, using the `initial` option, the observation itself can be what enables and disables analytics exclusively.